tests/test_runner/test_parallel.py

```diff
@@ -161,3 +202,81 @@ def test_add_duration(self):
         result = RemoteTestResult()
         result.addDuration(None, 2.3)
         self.assertEqual(result.collectedDurations, [("None", 2.3)])
+
+
+class ParallelTestSuiteTest(SimpleTestCase):
+    # NOTE: As of this writing there is no *integration* test that ensures that
+    #       RemoteTestResult includes the extra test_id element in the event
+    #       tuple, and that ParallelTestSuite correctly extracts and uses
+    #       test_id. (But there are related unit tests
+    #       RemoteTestResultTest.test_add_error_before_first_test and
+    #       ParallelTestSuiteTest.test_handle_add_error_before_first_test.)
+    def test_handle_add_error_before_first_test(self):
+        dummy_subsuites = []
+        pts = ParallelTestSuite(dummy_subsuites, processes=2)
+        result = TestResult()
+        test_id = "setUpClass (tests.test_foo.FooTest.setUpClass)"
+        test = TestCase()
+        err = _test_error_exc_info()
+        event = ("addError", -1, test_id, err)
+        pts.handle_event(result, tests=[test], event=event)
+
+        self.assertEqual(len(result.errors), 1)
+        actual_test, tb_and_details_str = result.errors[0]
+        self.assertIsInstance(actual_test, _ErrorHolder)
+        self.assertEqual(actual_test.id(), test_id)
+        self.assertRegex(
+            tb_and_details_str, re.compile(r"^Traceback .* woops.*", re.DOTALL)
+        )
```

Comment:
```suggestion
        self.assertIn("Traceback (most recent call last):", tb_and_details_str)
        self.assertIn("ValueError: woops", tb_and_details_str)
```
Instead of the regex calls I think we should do something like this

Reply:
Thank you, applied.